pkg/services/oauth/oauth.go

```diff
@@ -56,34 +95,196 @@ func (s *service) Configure() (err error) {
 		logrus.Info("Missing OAuth JWT signing key, generating new one")
 		s.crypto.GenerateKeyPair(s.oauthKeyEntity, false)
 	}
+
+	s.consent = nutsConsentClient.NewConsentStoreClient()
+
 	return
 }
 
 // CreateAccessToken extracts the claims out of the request, checks the validity and builds the access token
 func (s *service) CreateAccessToken(request services.CreateAccessTokenRequest) (*services.AccessTokenResult, error) {
-	// extract the JwtBearerToken
-	jwtBearerToken, err := s.parseAndValidateJwtBearerToken(request.RawJwtBearerToken)
-	if err != nil {
+	context := validationContext{
+		rawJwtBearerToken: request.RawJwtBearerToken,
+	}
+
+	// extract the JwtBearerToken, validates according to RFC003 §5.2.1.1
+	// also check if used algorithms are according to spec (ES*** and PS***)
+	// and checks basic validity. Set jwtBearerToken in validationContext
+	if err := s.parseAndValidateJwtBearerToken(&context); err != nil {
 		return nil, fmt.Errorf("jwt bearer token validation failed: %w", err)
 	}
 
-	// Validate the AuthTokenContainer
-	res, err := s.contractValidator.ValidateJwt(jwtBearerToken.AuthTokenContainer, request.VendorIdentifier)
-	if err != nil {
+	// check the maximum validity, according to RFC003 §5.2.1.4
+	if context.jwtBearerToken.ExpiresAt-context.jwtBearerToken.IssuedAt > OauthBearerTokenMaxValidity {
+		return nil, errors.New("JWT validity to long")
+	}
+
+	// check the actor against the registry, according to RFC003 §5.2.1.3
+	// checks signing certificate and sets vendor, actorName in validationContext
+	if err := s.validateIssuer(&context); err != nil {
```

Comment:
no error wrapping below, are errors descriptive enough in itself?

Reply:
it was a huge function with all validations that were wrapping something already. Logic has only be divided into multiple funcs to track if all requirements are implemented. So errors can be safely returned without wrapping.